ui/flow_of_allowing_to_add_models.md

```diff
@@ -0,0 +1,123 @@
+You can copy from another form and adjust as needed.
+
+Here you determine the GUI form the user will see when creating a new instance of your item. I won't go into the details of it in this guide.\
+When developing, it's safest to start with a minial form just to see that it's working.
+
+Note that the fields on your form will constitute the `options` argument for your `self.raw_create_your_item` method.
+
+After you add your form, you need to register it in `component-definitions-common.js`.
+
+## The UI Controller
+In your controller in `plugins/manageiq-ui-classic/app/controllers/your_item_controller.rb` you need to add the `new` method.\
+You can copy it from another controller and adjust as needed.
+
+Another thing you need in your controller is a method called `specific_buttons(pressed)`. If you don't have it already, copy it from another controller.\
+The method consists of a `case`.\
+Add the following `when` to the case. If you have just copied the method from another controller you can remove all other `when`s.
+```ruby
+when 'your_item_new'
+  javascript_redirect(:action => 'new')
+end
+```
+
+## haml template for 'new' operation
+Create `new.html.haml` under `plugins/manageiq-ui-classic/app/views/your_item/`. You can copy it from another item and adjust as needed.
+Make sure to point at your JS Form which you defined above `react 'YourItemForm'`
+
+## UI routes
+In `manageiq-ui-classic/config/routes.rb`, you need to find the section for your_item. Then you need to make sure there's `new` in the get section, and `button` in the post section.
+
+## API Controller
+If there isn't one, create a controller for your item in `manageiq-api/app/controllers/api/your_items_controller.rb`. You can copy one of the existing controllers and adjust to your item.
+
+Make sure that there is a `create_resource` method in the api controller. If not, copy if from some other controller and adjust to fit your item.\
+This method is supposed to call the method`create_your_item_queue` on the base application-record.
+
+An API controller with a good example of `create_resource` method is `auth_key_pairs_controller.rb`, so you can copy-paste from there and adjust as needed.
+
+If you are adding a new controller, you'll need to add routes to in `manageiq-api/api.yml`. Copy from one of the existing routes and adjust.
+
+### API Controller Specs
+The spec file needs to be in `plugins/manageiq-api/spec/requests/your_items_spec.rb`.\
+If a spec file doesn't exist for the api-controller, such as when you have just created a new api-controller, create a new one.  
+
+Add the following tests to the spec file. You can copy-paste them from other specs and adjust to your item.
+1. Good path where a POST succeeds in creating a new item.
+2. A create request without a valid ems-id should fail.
+3. A create request for a user without an appropriate role should fail.
```

Comment:
This would be a good place to mention the product_feature/identifier that you have to authorize the user for in the specs

Reply:
I've added a section about product features. Could you check it?